src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrVersionFileMerger.cs

```diff
@@ -0,0 +1,318 @@
+        var mergedJson = SimpleConfigJson.ApplyJsonChanges(currentJson, mergedChanges);
+
+        var newJson = new GitFile(targetRepo.Path / jsonRelativePath, mergedJson);
+        await _localGitRepoFactory.Create(targetRepo.Path).StageAsync(["."]);
+
+        await _gitRepoFactory.CreateClient(targetRepo.Path)
+            .CommitFilesAsync(
+                [newJson],
+                targetRepo.Path,
+                targetRepoCurrentRef,
+                $"Merge {jsonRelativePath} changes from VMR");
+    }
+
+    public async Task<VersionFileChanges<DependencyUpdate>> MergeVersionDetails(
+        Codeflow lastFlow,
+        Codeflow currentFlow,
+        string mappingName,
+        ILocalGitRepo targetRepo,
+        string targetBranch)
+    {
+        _logger.LogInformation(
+            "Resolving backflow dependency updates between VMR {vmrSha1}..{vmrSha2} and {repo} {repoSha1}..{repoSha2}",
+            lastFlow.VmrSha,
+            Constants.HEAD,
+            mappingName,
+            lastFlow.RepoSha,
+            targetBranch);
+
+        var previousRepoDependencies = await GetRepoDependencies(targetRepo, lastFlow.RepoSha);
+        var currentRepoDependencies = await GetRepoDependencies(targetRepo, targetBranch);
+
+        // Similarly to the code flow algorithm, we compare the corresponding commits
+        // and the contents of the version files inside.
+        // We distinguish the direction of the previous flow vs the current flow.
+        var vmr = _localGitRepoFactory.Create(_vmrInfo.VmrPath);
+        var previousVmrDependencies = lastFlow is Backflow
+            ? await GetVmrDependencies(vmr, mappingName, lastFlow.VmrSha)
+            : previousRepoDependencies;
+        var currentVmrDependencies = await GetVmrDependencies(vmr, mappingName, currentFlow.VmrSha);
+
+        List<DependencyUpdate> repoChanges = ComputeChanges(
+            previousRepoDependencies,
+            currentRepoDependencies);
+
+        List<DependencyUpdate> vmrChanges = ComputeChanges(
+            previousVmrDependencies,
+            currentVmrDependencies);
+
+        var mergedChanges = MergeDependencyUpdates(repoChanges, vmrChanges);
+
+        await ApplyVersionDetailsChangesAsync(targetRepo.Path, mergedChanges);
+
+        return mergedChanges;
+    }
+
+    private VersionFileChanges<JsonVersionProperty> MergeJsonChanges(
+        List<JsonVersionProperty> repoChanges,
+        List<JsonVersionProperty> vmrChanges)
+        => MergeVersionFileChanges(repoChanges, vmrChanges);
+
+    private VersionFileChanges<DependencyUpdate> MergeDependencyUpdates(
+        List<DependencyUpdate> repoChanges,
+        List<DependencyUpdate> vmrChanges)
+        => MergeVersionFileChanges(repoChanges, vmrChanges);
+
+    private VersionFileChanges<T> MergeVersionFileChanges<T>(
+        List<T> repoChanges,
+        List<T> vmrChanges) where T : IVersionFileProperty
```

Comment:
Rather than using the registry, pass the comparer methods as a generic argument here.
```suggestion
    private VersionFileChanges<T> MergeVersionFileChanges<T>(
        List<T> repoChanges,
        List<T> vmrChanges,
        Func<T, T, T> propertySelector) where T : IVersionFileProperty
```